synchronization/src/main/java/de/cyface/synchronization/NetworkCallback.java

```diff
@@ -54,13 +55,28 @@
         this.currentSynchronizationAccount = currentSynchronizationAccount;
     }
 
+    @Override
+    public void onLost(@NonNull final Network network) {
+
+        // This is required for < MINIMUM_VERSION_TO_USE_NOT_METERED_FLAG or else we are not informed about a lost wifi
+        // connection e.g. on Android 6.0.1 (MOV-650, possibly also MOV-645)
+
+        Log.v(TAG, "NetworkCallback.onLost: setConnected to false.");
+        surveyor.setConnected(false);
+    }
+
     @Override
     public void onCapabilitiesChanged(@NonNull final Network network, @NonNull final NetworkCapabilities capabilities) {
 
         // Ensure this event is only triggered for not metered connections when syncOnUnMeteredNetworkOnly
         if (surveyor.isSyncOnUnMeteredNetworkOnly()) {
-            final boolean unMeteredNetwork = capabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_NOT_METERED);
-            Validate.isTrue(unMeteredNetwork);
+
+            final boolean isUsingUnMeteredCheckInsteadOfWifi = Build.VERSION.SDK_INT >= MINIMUM_VERSION_TO_USE_NOT_METERED_FLAG;
+            if (isUsingUnMeteredCheckInsteadOfWifi) {
+                Validate.isTrue(capabilities.hasCapability(NetworkCapabilities.NET_CAPABILITY_NOT_METERED));
+            } else {
+                Validate.isTrue(capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI));
```

Comment:
This still crashes everything on a connection loss, doesn't it?

Reply:
No, because we only register to NetworkCallbacks with those attributes (see below).

Those validations just makes sure this "Filter" actually works as expected because I had a bug before where I accidentally used another filter in the callback registration than what I was expecting in here:

Here (or in the code below) the Registration:
```
final NetworkRequest.Builder requestBuilder = new NetworkRequest.Builder();
            if (syncOnUnMeteredNetworkOnly) {	          
                Log.v(TAG, "startSurveillance for wifi networks only");	                
               final boolean isUsingUnMeteredCheckInsteadOfWifi = Build.VERSION.SDK_INT >= MINIMUM_VERSION_TO_USE_NOT_METERED_FLAG;	
                  if (isUsingUnMeteredCheckInsteadOfWifi) {
                    requestBuilder.addCapability(NET_CAPABILITY_NOT_METERED);
                } else {
                    requestBuilder.addTransportType(NetworkCapabilities.TRANSPORT_WIFI);
                }
```